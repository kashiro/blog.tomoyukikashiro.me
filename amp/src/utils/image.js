export const headerBgUrl = (day) => {
  return day ? `/images/${ day % 7 }.jpg` : `/images/home.jpg`
}

export const headerBgClass = (day) => {
  return `header__bg_${ day % 7 }`
}
