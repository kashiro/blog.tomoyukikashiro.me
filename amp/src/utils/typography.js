import Typography from 'typography'
import typographyThemeKirkham from 'typography-theme-kirkham'

const typography = new Typography(typographyThemeKirkham)

// Hot reload typography in development.
if (process.env.NODE_ENV !== 'production') {
  typography.injectStyles()
}

export default typography
export const rhythm = typography.rhythm
export const scale = typography.scale
