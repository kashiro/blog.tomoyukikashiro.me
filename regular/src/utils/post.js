// import moment from 'moment'

export default class Post {

  static LANGS = {
    JA: 'ja',
    EN: 'en-US'
  }

  constructor(node) {
    this.node = node
  }
  
  get key() {
    return `${this.lang}-${this.slug}`
  }

  get title() {
    return this.node.frontmatter.title.trim()
  }

  get date() {
    if (!this.node.frontmatter.date) return
    return new Date(this.node.frontmatter.date.trim())
  }
  
  get type() {
    return 'article'
  }
  
  get isoDate() {
    return this.date.toISOString()
  }
  
  get formatDate() {
    return this.date.toLocaleString('en-US', { weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' })
    // return moment(this.date.toISOString()).format('dddd LL')
  }
  
  get summary() {
    return this.node.frontmatter.summary ? this.node.frontmatter.summary.trim() : ''
  }
  
  get slug() {
    return this.node.frontmatter.slug.trim()
  }
  
  get lang() {
    return this.node.frontmatter.lang.trim()
  }
  
  get isEn() {
    return this.lang === Post.LANGS.EN
  }
  
  get isJa() {
    return this.lang === Post.LANGS.JA
  }
  
  get alternativeLang() {
    return this.isEn ? Post.LANGS.JA : Post.LANGS.EN
  }
  
  get tags() {
    if (this.node.frontmatter.tags) {
      return this.node.frontmatter.tags.map(tag => tag.trim().toLowerCase()) 
    } else {
      return []
    }
  }
  
  get hasTags() {
    return !!this.tags.length
  }

  path(lang) {
    const langPath = (lang || this.lang) === Post.LANGS.JA ? '/ja/' : '/'
    return `/post${langPath}${this.slug}`
  }
  
  get alternativeLangPath() {
    return this.path(this.alternativeLang)
  }
  
  get html() {
    return this.node.html
  }
  
  get canonicalUrl() {
    return this.node.frontmatter.canonicalUrl
  }
}
  

