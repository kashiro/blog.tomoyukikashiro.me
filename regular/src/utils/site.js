import Post from './post'

export default class Site {

  constructor(siteMetadata) {
    this.meta = siteMetadata
  }
  
  // --- title ---
  
  get title() {
    return this.meta.title
  }
  
  get tagsPageTitle() {
    return `TAGS | ${this.title}`
  }
  
  tagPagePrefix(tag, totalCount) {
    return `Post${totalCount === 1 ? "" : "s"} tagged with "${tag.toUpperCase()}"`
  }
  
  tagPageTitle(tag, totalCount) {
    return `${this.tagPagePrefix(tag, totalCount)} TAG | ${this.title}`
  }

  // --- description ---
  get description() {
    return this.meta.description
  }
  
  get tagsPageDescription() {
    return `TAGS | ${this.meta.description}`
  }

  tagPageDescription(tag, totalCount) {
    return `${this.tagPagePrefix(tag, totalCount)} TAG | ${this.description}`
  }
  
  postPageDescription(post) {
    return post.summary
  }
  
  
  // --- url ---
  
  get url() {
    return this.meta.siteUrl
  }
  
  get ampUrl() {
    return this.meta.ampUrl
  }
  
  postUrl(post, lang) {
    return `${this.url}${post.path(lang)}/`
  }
  
  postEnUrl(post) {
    return this.postUrl(post, Post.LANGS.EN)
  }
  
  postAlternativeLangUrl(post) {
    return this.postUrl(post, post.alternativeLang)
  }
  
  canonicalPostUrl(post, lang) {
    if (post.canonicalUrl) return post.canonicalUrl
    return this.postUrl(post, lang)
  }
  
  canonicalPostAmpUrl(post) {
    return `${this.ampUrl}${post.path()}/`
  }

  get tagsPageUrl() {
    return `${this.url}/tags`
  }
  
  tagPageUrl(tag) {
    return `${this.url}/tag/${tag.toLowerCase()}/`
  }
  
  get profileUrl() {
    return this.meta.profileUrl
  }
  
  // --- others ---

  get type() {
    return 'website'
  }

  get author() {
    return this.meta.author
  }
  
  get twitterUserName() {
    return this.meta.twitterUserName
  }
}
