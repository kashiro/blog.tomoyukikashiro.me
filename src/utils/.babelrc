{
  "presets": [
    [
      "@babel/preset-env",
      {
        "loose": true,
        "modules": "commonjs",
        "useBuiltIns": "usage",
        "targets": {
          "node": true
        }
      }
    ]
  ],
  "plugins": [
    [
      "@babel/plugin-proposal-class-properties",
      {
        "loose": true
      }
    ]
  ]
}
