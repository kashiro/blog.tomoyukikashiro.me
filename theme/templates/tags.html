{% extends "base.html" %}
{% import "macros/meta.html" as meta_macros with context %}
{% import "macros/contents.html" as contents_macros with context %}
{% import "macros/json_ld.html" as json_ld_macros with context %}

{% set title %}TAGS | {{ SITENAME }}{% endset %}
{% set description %}TAGS | {{ SITENAME }}{% endset %}
{% set url %}{{ SITEURL }}/{{ TAGS_URL }}{% endset %}
{% set image %}{{ contents_macros.header_bg_url() }}{% endset %}

{% block title %}{{ title }}{% endblock title %}

{% block meta %}
  {{ super() }}
  <meta name="description" content="{{ description }}">
  {{ meta_macros.og_metas(title=title, description=description, type='website', url=url, image=image) }}
  <link rel="canonical" href="{{ url }}">
{% endblock %}

{% block json_ld %}
  {{ super() }}
  {{ json_ld_macros.tag_bread_crumb() }}
{% endblock %}

{% block content %}
  <main>
    <header class="l-header header header__bg-home">
      <div class="header__container">
        <h1 class="header__title"><a class="link-no-decoration" href="{{ url }}">{{ SITENAME }}<br>TAGS</a></h1>
      </div>
    </header>
    <div class="body">
      {%- for tag, articles in tags|sort %}
        <section class="section">
          <h1 class="section__title"><i class="material-icons section__title-icon">label_outline</i><a href="/{{ tag.url }}">{{ tag|upper }}</a> ({{ articles|count }})</h1>
          {% for article in articles %}
            {{ contents_macros.article_item(article) }}
          {% endfor %}
        </section>
      {% endfor %}
    </div>
  </main>
{% endblock content %}
