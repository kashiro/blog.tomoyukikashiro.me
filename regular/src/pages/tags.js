import React from "react"
import Helmet from 'react-helmet'
import { Link, graphql } from "gatsby"
import Layout from '../components/Layout'
import TagsBreadCrumb from '../components/ld_json/TagsBreadCrumb'
import MetaSocial from '../components/MetaSocial'
import {headerBgUrl} from '../utils/image'
import Header from '../components/Header'

import HeaderStyles from '../components/Header.module.css'
import LabelSvg from '../assets/images/label.svg'
import Site from '../utils/site'

export const TagsPageHead = ({ site }) => (
  <React.Fragment>
    <Helmet>
      <title>{ site.tagsPageTitle }</title>
      <meta name="description" content={ site.tagsPageDescription }/>
      <link rel="canonical" href={ `${site.tagsPageUrl}/` } />
    </Helmet>
    <MetaSocial
      site={ site }
      title={ site.tagsPageTitle }
      description={ site.tagsPageDescription }
      type={ site.type }
      url={ `${site.tagsPageUrl}/` }
      image={ headerBgUrl() }
    />
    <TagsBreadCrumb site={site} />
  </React.Fragment>
)

const TagsPage = ({
  data: {
    allMarkdownRemark: { group },
    site: { siteMetadata }
  },
}) => {
  const site = new Site(siteMetadata)
  return (
    <Layout site={site}>
      <TagsPageHead site={site} /> 
      <main>
        <Header klass="header__bg_home" text={ site.title } link="/tags/">
          <h2 className={HeaderStyles.header__subtitle}>TAGS</h2>
        </Header>
        <div className="body">
          {group.map(tag => (
            <section className="section" key={ tag.fieldValue }>
              <h1 className="section__title">
                <i><LabelSvg/></i>
                <Link to={ `/tag/${tag.fieldValue.toLowerCase()}/ `}>{ tag.fieldValue.toUpperCase() }</Link> ({ tag.totalCount })
              </h1>
            </section>
          ))}
        </div>
      </main>
    </Layout>
  )
}

export default TagsPage

export const pageQuery = graphql`
  query {
    site {
      siteMetadata {
        title
        author
        description
        siteUrl
        profileUrl
        twitterUserName
        ampUrl
      }
    }
    allMarkdownRemark(
      limit: 2000
    ) {
      group(field: frontmatter___tags) {
        fieldValue
        totalCount
      }
    }
  }
`
