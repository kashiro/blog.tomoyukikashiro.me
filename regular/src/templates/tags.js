import React from "react"
import Layout from '../components/Layout'
import Helmet from 'react-helmet'
// Components
import { Link, graphql } from "gatsby"
import TagsBreadCrumb from '../components/ld_json/TagsBreadCrumb'
import MetaSocial from '../components/MetaSocial'
import { headerBgUrl } from '../utils/image'
import Article from '../components/Article'
import Header from '../components/Header'
import HeaderStyles from '../components/Header.module.css'

const Tags = ({ pageContext, data }) => {
  const { tag } = pageContext
  const { edges, totalCount } = data.allMarkdownRemark
  const { title, siteUrl } = data.site.siteMetadata
  const tagHeader = `Post${
    totalCount === 1 ? "" : "s"
    } tagged with "${tag}"`

  return (
    <Layout>
      <Helmet>
        <title>{ `${tagHeader} | ${title}` }</title>
        <meta name="description" content={ `${tagHeader} | ${title}` } />
        <link rel="canonical" href={ `/tag/${tag.toLowerCase()}/` } />
      </Helmet>
      <MetaSocial
        title={ `${ tag.toUpperCase() } TAG | ${ title }` }
        description={ `${ tag.toUpperCase() } TAG | ${ title }` }
        type="website"
        url={ `${ siteUrl }/tag/${ tag.toLowerCase() }/` }
        image={ headerBgUrl() }
      />
      <TagsBreadCrumb tag={tag} />
      <main>
        <Header klass="header__bg_home" text={ title } link={`${ siteUrl }/tag/${ tag.toLowerCase() }/`}>
          <i className={ `material-icons ${HeaderStyles.header__title_icon}` }>label</i>{ tag.toUpperCase() } TAG
        </Header>
        <div className="body">
           { edges.map(({ node }) => <Article node={node} key={node.frontmatter.slug} />)}
        </div>
      </main>
    </Layout>
  )
}

export default Tags

export const pageQuery = graphql`
  query($tag: String) {
    site {
      siteMetadata {
        title
      }
    }
    allMarkdownRemark(
      limit: 2000
      sort: { fields: [frontmatter___date], order: DESC }
      filter: { frontmatter: { tags: { in: [$tag] } } }
    ) {
      totalCount
      edges {
        node {
          excerpt
          frontmatter {
            title
            slug
            date
          }
        }
      }
    }
  }
`
