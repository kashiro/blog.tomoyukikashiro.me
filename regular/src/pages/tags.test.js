import React from 'react'
import renderer from 'react-test-renderer'
import { requestAnimationFrame } from '../../__tests__/utils'
import { TagsPageHead } from './tags'

describe('TagsPageHead', function () {
  const site = {
    tagsPageTitle: 'test title',
    tagsPageDescription: 'test description',
    tagsPageUrl: 'https://example.com/tags',
    url: 'https://example.com',
    type: 'website',
    twitterUserName: 'testAccount' 
  }
  let headElement;
  beforeEach(() => {
    headElement =
      headElement || document.head || document.querySelector("head");

    // resets DOM after each run
    headElement.innerHTML = "";
  })
  it('TagsPageHead render correctly', function (done) {
    renderer.create(<TagsPageHead site={ site }/>).toJSON()
    requestAnimationFrame(() => {
      const tree = renderer.create(headElement.innerHTML).toJSON()   
      expect(tree).toMatchSnapshot()
      done()
    })
  }) 
})
