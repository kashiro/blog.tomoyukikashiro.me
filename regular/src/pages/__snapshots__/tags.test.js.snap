// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagsPageHead TagsPageHead render correctly 1`] = `
"<title>test title</title><link rel=\\"canonical\\" href=\\"https://example.com/tags/\\" data-react-helmet=\\"true\\"><meta name=\\"description\\" content=\\"test description\\" data-react-helmet=\\"true\\"><meta property=\\"og:title\\" content=\\"test title\\" data-react-helmet=\\"true\\"><meta property=\\"og:type\\" content=\\"website\\" data-react-helmet=\\"true\\"><meta property=\\"og:image\\" content=\\"https://example.com/images/home.jpg\\" data-react-helmet=\\"true\\"><meta property=\\"og:url\\" content=\\"https://example.com/tags/\\" data-react-helmet=\\"true\\"><meta property=\\"og:description\\" content=\\"test description\\" data-react-helmet=\\"true\\"><meta property=\\"og:locale\\" content=\\"en\\" data-react-helmet=\\"true\\"><meta property=\\"og:site_name\\" content=\\"\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:card\\" content=\\"summary_large_image\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:creator\\" content=\\"@testAccount\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:title\\" content=\\"test title\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:description\\" content=\\"test description\\" data-react-helmet=\\"true\\"><meta name=\\"twitter:image\\" content=\\"https://example.com/images/home.jpg\\" data-react-helmet=\\"true\\"><script type=\\"application/ld+json\\" data-react-helmet=\\"true\\">{
           \\"@context\\": \\"http://schema.org\\",
           \\"@type\\": \\"BreadcrumbList\\",
           \\"itemListElement\\": [
             {
               \\"@type\\": \\"ListItem\\",
               \\"position\\": 1,
               \\"item\\": {
                 \\"@id\\": \\"https://example.com/tags/\\",
                 \\"name\\": \\"tags\\",
                 \\"image\\": \\"https://example.com/images/home.jpg\\"
               }
             }
           ]
        }</script>"
`;
